crates/ouroboros-traits/src/has_stake_distribution/mod.rs

```diff
@@ -1,3 +1,46 @@
+// Copyright 2024 PRAGMA
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use amaru_kernel::{Lovelace, PoolId, Slot, VrfKeyhash};
+
 pub mod mock;
-mod traits;
-pub use traits::*;
+
+#[derive(Debug)]
+pub struct PoolSummary {
+    /// The blake2b-256 hash digest of the pool's VRF public key.
+    pub vrf: VrfKeyhash,
+    /// Total stake, in Lovelace, delegated to registered pools.
+    pub active_stake: Lovelace,
```

Comment:
Is this not a value that's common to _all_ pools? If so, why put it into each pool structure?

Reply:
(note: I didn't alter this module in any ways, just moved it up one layer)

It isn't really that the value is in each "pool structure", it is in the return type of `getPool`. It is simply for convenience since this information is readily available (and required) when fetching. 

We could also just separate both having `get_pool` and `get_active_stake`, but that means doing double work behind the scene. One could say "premature optimization", one could also say "wrong function name". But calling is "get_pool_info_for_validation" felt somewhat wrong.